### ISSUE TITLE
textureLod conversion seems off

### ISSUE THREAD

**Issue by cybik** [LukasBanana/XShaderCompiler#27]
`float4 depth_ao = tex2Dlod(a_thing, float4(tex, 0.0f, 0.0f));` gets converted to `vec4 depth_ao = textureLod(Ssao, vec4(xsv_TEXCOORD0, 0.0f, 0.0f));`

The translation doesn't validate:
```
ERROR: 0:23: 'textureLod' : no matching overloaded function found 
ERROR: 0:23: '=' :  cannot convert from 'const float' to 'temp 4-component vector of float'
ERROR: 0:23: '' :  syntax error
ERROR: 3 compilation errors.  No code generated.
```

**Comment by LukasBanana** [comment]
Thanks for the hint, I haven't tested the compiler much with Shader Model 3 shaders.
So this is erroneously ignored in the converter so far.

**Comment by LukasBanana** [comment]
Fixed with [2aa19f1](https://github.com/LukasBanana/XShaderCompiler/commit/2aa19f12a782ffe4d1e889f5056b352deb2fea29)

Currently, the conversion for these intrinsics always creates a temporary varibale, but maybe I can enhance the output when the temporary variable is avoidable.
